File: servers/scalatra/src/main/scala/com/wordnik/client/model/instrument.scala
package com.wordnik.client.model

import java.util.Date
case class Instrument (
  symbol: String,

  rootSymbol: Option[String],

  state: Option[String],

  typ: Option[String],

  listing: Option[Date],

  front: Option[Date],

  expiry: Option[Date],

  inverseLeg: Option[String],

  sellLeg: Option[String],

  buyLeg: Option[String],

  underlying: Option[String],

  quoteCurrency: Option[String],

  underlyingSymbol: Option[String],

  reference: Option[String],

  referenceSymbol: Option[String],

  tickSize: Option[Double],

  multiplier: Option[Double],

  settlCurrency: Option[String],

  underlyingToSettleMultiplier: Option[Double],

  quoteToSettleMultiplier: Option[Double],

  isQuanto: Option[Boolean],

  isInverse: Option[Boolean],

  initMargin: Option[Double],

  maintMargin: Option[Double],

  limit: Option[Double],

  makerFee: Option[Double],

  takerFee: Option[Double],

  insuranceFee: Option[Double],

  openingTimestamp: Option[Date],

  closingTimestamp: Option[Date],

  prevClosePrice: Option[Double],

  limitDownPrice: Option[Double],

  limitUpPrice: Option[Double],

  prevTotalVolume: Option[Double],

  totalVolume: Option[Double],

  volume: Option[Double],

  vwap: Option[Double],

  highPrice: Option[Double],

  lowPrice: Option[Double],

  lastPrice: Option[Double],

  lastTickDirection: Option[String],

  lastChangePcnt: Option[Double],

  bidPrice: Option[Double],

  midPrice: Option[Double],

  askPrice: Option[Double],

  openInterest: Option[Double],

  settledPrice: Option[Double],

  timestamp: Option[Date]

  )

File: clients/typescript-angular/API/Client/Instrument.ts
/// <reference path="api.d.ts" />

namespace API.Client {
    'use strict';

    export interface Instrument {

        symbol: string;

        rootSymbol?: string;

        state?: string;

        typ?: string;

        listing?: date;

        front?: date;

        expiry?: date;

        relistInterval?: date;

        inverseLeg?: string;

        sellLeg?: string;

        buyLeg?: string;

        underlying?: string;

        quoteCurrency?: string;

        underlyingSymbol?: string;

        reference?: string;

        referenceSymbol?: string;

        tickSize?: number;

        multiplier?: number;

        settlCurrency?: string;

        underlyingToSettleMultiplier?: number;

        quoteToSettleMultiplier?: number;

        isQuanto?: boolean;

        isInverse?: boolean;

        initMargin?: number;

        maintMargin?: number;

        limit?: number;

        capped?: boolean;

        taxed?: boolean;

        makerFee?: number;

        takerFee?: number;

        insuranceFee?: number;

        hedgerMakerFee?: number;

        hedgerTakerFee?: number;

        hedgerInsuranceFee?: number;

        openingTimestamp?: date;

        closingTimestamp?: date;

        sessionInterval?: date;

        prevClosePrice?: number;

        limitDownPrice?: number;

        limitUpPrice?: number;

        bankruptLimitDownPrice?: number;

        bankruptLimitUpPrice?: number;

        prevTotalVolume?: number;

        totalVolume?: number;

        volume?: number;

        volume24h?: number;

        prevTotalTurnover?: number;

        totalTurnover?: number;

        turnover?: number;

        turnover24h?: number;

        prevPrice24h?: number;

        vwap?: number;

        highPrice?: number;

        lowPrice?: number;

        lastPrice?: number;

        lastPriceProtected?: number;

        lastTickDirection?: string;

        lastChangePcnt?: number;

        bidPrice?: number;

        midPrice?: number;

        askPrice?: number;

        impactBidPrice?: number;

        impactMidPrice?: number;

        impactAskPrice?: number;

        hasLiquidity?: boolean;

        openInterest?: number;

        openValue?: number;

        fairMethod?: string;

        fairBasisRate?: number;

        fairBasis?: number;

        fairPrice?: number;

        markMethod?: string;

        markPrice?: number;

        indicativeTaxRate?: number;

        indicativeSettlePrice?: number;

        settledPrice?: number;

        timestamp?: date;
    }

}
